### DISCUSSION TITLE
vitest: Resetting state between tests in react-dom

### DISCUSSION THREAD

**Discussion by kaplanelad** [pmndrs/zustand#1829] [Q&A] [answered]
I'm currently using vitest as a test library to test my application. I want to reset the state between each test run. I attempted to follow the instructions provided in the documentation (https://docs.pmnd.rs/zustand/guides/testing), but unfortunately, it didn't work as expected. 
Interestingly, when I used Jest for testing, the state reset worked properly. Is there any possibility that you could provide an example of how to implement the state reset between tests using vitest?

**Comment by dbritto-dev** [comment]
@kaplanelad did you read this https://vitest.dev/guide/migration.html#?

**Comment by kaplanelad** [comment]
@dbritto-dev thanks for the fast response. 
Yes, i tried to do it and is seems to be not working

change from 
```
const { create: actualCreate } =
  jest.requireActual<typeof import('zustand')>('zustand')
```
to
```
const { create: actualCreate } =
  await vi.importActual<typeof import('zustand')>('zustand')
```

Added the setup file into `setupFiles` and validate that the file is loaded
but the the function `create` function is't called

**Comment by dbritto-dev** [comment]
@kaplanelad would you mind sharing us a basic example of what you are doing using codesandox?

**Comment by kaplanelad** [comment]
@dbritto-dev, 
I have attached an example for you here:https://stackblitz.com/edit/zustand-with-vitest-reset-store

Here's some additional context for better understanding:
I have created a simple store located at src/store.ts. This store implements a function called usetestStore, which allows you to add a given number to an array.

To test the functionality, I created a file named test/store.test.ts. Inside this file, there are two tests. Each test utilizes the add function. The expected outcome is that the second test does not include the data that was added in the first test.

In your vite.config.ts file, you have configured the setupFiles option. Upon examining the __mocks__/zustand.ts file, I noticed that the create function is not being invoked, resulting in the storeResetFns array being empty.

I hope this provides the information you were looking for. Let me know if you need any further assistance!

thanks for your help!

**Reply by dbritto-dev** [reply]
@kaplanelad that works I'll look into that

**Reply by kaplanelad** [reply]
thanks @dbritto-dev, waiting for your investigation

**Reply by dbritto-dev** [reply]
@kaplanelad two things you need to do.

- Remove `__mocks__/zustand.ts` from you `setupFiles` on your `vitest.config.ts`
- On `vitest.config.ts` you need to set up `setup-tests.ts` on `setupFiles` and inside of `setup-tests.ts` you need to import vitest and mock zustand to make it work like in `jest`

```ts
import { vi } from 'vitest'

vi.mock('zustand')
```

Please keep this in mind

```
WARNING

Beware that if you don't call vi.mock, modules are not mocked automatically. To replicate Jest's automocking behaviour, you can call vi.mock for each required module inside [setupFiles](https://vitest.dev/config/#setupfiles).
```

**Reply by dbritto-dev** [reply, marked as answer]
@kaplanelad here you go, I updated your demo with my suggestions -> https://stackblitz.com/edit/zustand-with-vitest-reset-store-2vqbyq?file=setup-tests.ts

**Reply by kaplanelad** [reply]
Thank you, @dbritto-dev, for your assistance. I appreciate your time.
I will check your suggestion to my application. Many thanks!
